github_dynamic_targets/main.tf

```diff
@@ -0,0 +1,177 @@
+provider "sym" {
+  org = "sym-example"
+}
+
+############ General AWS Secrets Manager Setup ##############
+
+# Creates an AWS IAM Role that the Sym Runtime can use for execution
+# Allow the runtime to assume roles in the /sym/ path in your AWS Account
+module "runtime_connector" {
+  source  = "terraform.symops.com/symopsio/runtime-connector/sym"
+  version = ">= 1.1.0"
+
+  # The aws/secretsmgr addon is required to access secrets
+  addons = ["aws/secretsmgr"]
+
+  environment = "main"
+}
+
+# An Integration that tells the Sym Runtime resource which AWS Role to assume
+# (The AWS Role created by the runtime_connector module)
+resource "sym_integration" "runtime_context" {
+  type = "permission_context"
+  name = "main-runtime"
+
+  external_id = module.runtime_connector.settings.account_id
+  settings    = module.runtime_connector.settings
+}
+
+# This resource tells Sym which role to use to access your AWS Secrets Manager
+resource "sym_secrets" "this" {
+  type = "aws_secrets_manager"
+  name = "main-sym-secrets"
+
+  settings = {
+    context_id = sym_integration.runtime_context.id
+  }
+}
+
+############ GitHub Integration and Secret Setup ##############
+
+# An AWS Secrets Manager Secret to hold your GitHub Access Token. Set the value with:
+# aws secretsmanager put-secret-value --secret-id "main/github-access-token" --secret-string "YOUR-GITHUB-ACCESS-TOKEN"
+resource "aws_secretsmanager_secret" "github_access_token" {
+  name        = "main/github-access-token"
+  description = "API Key for Sym to call GitHub APIs"
+
+  tags = {
+    # This SymEnv tag is required and MUST match the `environment` in your `runtime_connector` module
+    # because the aws/secretsmgr only grants access to secrets tagged with a matching SymEnv value
+    SymEnv = "main"
+  }
+}
+
+# This resource tells Sym how to access your GitHub Access Key.
+resource "sym_secret" "github_access_token" {
+  # The source of your secrets and the permissions needed to access
+  # i.e. AWS Secrets Manager, access with IAM Role.
+  source_id = sym_secrets.this.id
+
+  # Name of the key in AWS Secrets Manager
+  path = aws_secretsmanager_secret.github_access_token.name
+}
+
+# The GitHub Integration that your Sym Strategy uses to manage your GitHub Repo targets
+resource "sym_integration" "github" {
+  type = "github"
+  name = "main-github-integration"
+
+  # The external ID is your GitHub Organization name
+  external_id = "sym-test"
+
+  settings = {
+    # `type=github` sym_integrations have a required setting `api_token_secret`,
+    # which must point to a sym_secret referencing your GitHub Access Token
+    api_token_secret = sym_secret.github_access_token.id
+  }
+}
+
+############ GitHub Strategy Setup ##############
+
+# A target GitHub repo that your Sym Strategy can manage access to
+resource "sym_target" "private-repo" {
+  type = "github_repo"
+
+  name = "private-repos"
+  label = "Private Repos"
+
+  field_bindings = ["repo_name"]
```

Comment:
Probably want a comment explaining what this is?

Reply:
ya sorry just added it